### PR TITLE
add dolt_hashof_table func and dolt_hashof (Deprecate hashof)

### PR DESCRIPTION
**Pull request by bheni**
I created this function for my GDC demo.  We can talk about pros and cons of integrating this back to main.

### CODE REVIEW

**Comment by coffeegoddd** [comment]
@bheni __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>ff57f97</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>ff57f97</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

**Comment by coffeegoddd** [comment]
@coffeegoddd __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>c84fc04</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>c84fc04</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

File: go/libraries/doltcore/sqle/dfunctions/hashof_table.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+	"fmt"
+
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/expression"
+	"github.com/dolthub/go-mysql-server/sql/types"
+
+	"github.com/dolthub/dolt/go/libraries/doltcore/sqle/dsess"
+)
+
+const HashOfTableFuncName = "hashof_table"
```
Comment (fulghum) [inline go/libraries/doltcore/sqle/dfunctions/hashof_table.go:28]: Maybe something like `dolt_table_hash` or something? Kinda nice to have the common `dolt` prefix. `hashof` always felt a little out of place to me without that.

File: go/libraries/doltcore/sqle/dfunctions/hashof_table.go [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+	if err != nil {
+		return nil, fmt.Errorf("error getting hash of table %s: %w", tableName, err)
+	}
+
+	return h.String(), nil
+}
+
+// String implements the Stringer interface.
+func (t *HashOfTable) String() string {
+	return fmt.Sprintf("HASHOF(%s)", t.Child.String())
```
Comment (fulghum) [inline go/libraries/doltcore/sqle/dfunctions/hashof_table.go:84]: (minor) you could plug in `HashOfTableFuncName` here.

```suggestion
	return fmt.Sprintf("%s(%s)", HashOfTableFuncName, t.Child.String())
```

File: go/libraries/doltcore/sqle/dfunctions/hashof_table.go [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+}
+
+// FunctionName implements the FunctionExpression interface
+func (t *HashOfTable) FunctionName() string {
+	return HashOfTableFuncName
+}
+
+// Description implements the FunctionExpression interface
+func (t *HashOfTable) Description() string {
+	return "returns the commit hash of a branch or other commit spec"
```
Comment (fulghum) [inline go/libraries/doltcore/sqle/dfunctions/hashof_table.go:94]:
```suggestion
	return "returns a hash of the contents of a table, typically used for detecting if a table's data has changed"
```

**Review by fulghum — APPROVED** [review]
Looks like a useful function to provide! Just one suggestion on a name.

**Comment by coffeegoddd** [comment]
@bheni __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>3e98d49</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>3e98d49</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

**Review by fulghum — APPROVED** [review]
🚢 🙌

**Pull request merged**